graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -277,4 +293,23 @@ def enter_FieldDefinition(self, node, *args):
         if self.in_query_type:
             field_name = node.name.value
             new_field_name = self.renamings.get(field_name, field_name)  # Default use original
-            node.name.value = new_field_name
+            if new_field_name == field_name:
+                return None
+            else:  # Make copy of node with the changed name, return the copy
+                field_node_with_new_name = _get_copy_of_node_with_new_name(node, new_field_name)
+                return field_node_with_new_name
+
+
+def _get_copy_of_node_with_new_name(node, new_name):
+    """Return a node with new_name as its name and otherwise identical to the input node.
+
+    Args:
+        node: type Node, with a .name attribute. Not modified by this function
+        new_name: str, name to give to the output node
+    """
+    node_with_new_name = copy(node)  # shallow copy is enough
+    try:
+        node_with_new_name.name = ast_types.Name(value=new_name)
+    except AttributeError:
```

Comment:
Can we switch this into an ahead-of-time assertion that the type of `node` is one of the expected types (e.g. `Field` or whatever it is expected to be)?

The reason I prefer this is that `AttributeError` could in principle be raised in a few different places -- two in this case, but potentially more if we ever refactor this code: `node_with_new_name.name` and `ast_types.Name`. This makes it difficult to be sure that the error we _think_ we're catching is actually the error that happened.